### PR TITLE
unify special page messaging

### PR DESCRIPTION
**Pull request by shakyShane**
https://app.asana.com/0/1201614831475344/1207244302971195/f

### TL;DR

Updated messaging setup in special pages to use a shared `createSpecialPageMessaging` function.

### What changed?

- Refactored messaging setup in multiple special pages
- Created a shared createSpecialPageMessaging function
- NOTE: no functionality is changed in this PR.

### How to test?

- all changed pages have integration tests to verify

### Why make this change?

Centralized messaging setup for consistency and reusability

---

### CODE REVIEW

**Comment by shakyShane** [comment]
* **#957** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/957?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shakyShane and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/special-pages/pages/onboarding/app/messages.js [hunk 1]
```diff
@@ -51,6 +47,15 @@ export class OnboardingMessages {
...
     async init () {
+        if (this.injectName === "integration") {
+            return {
+                stepDefinitions: {
+                    systemSettings: {
+                        rows: ["dock", "import", "default-browser"]
+                    }
+                }
+            }
+        }
```
Comment (shakyShane) [inline packages/special-pages/pages/onboarding/app/messages.js:58]: This is an example of bringing a simple default value into the correct place in the application - it's the minimal needed in this example to allow local debugging 

This kind of situation was highlighted recently with other contributors - in short we need to ensure the connection points to the browser don't prevent local development

**Review by shakyShane — COMMENTED** [review]
@greyivy  in the spirit of merging often and early, could you review this first PR, which is just a small refactoring to set us up nicely for the NTP

**Review by greyivy — APPROVED** [review]
LGTM!

**Pull request merged**